src/tir/transforms/storage_rewrite.cc

```diff
@@ -504,8 +503,13 @@ class StoragePlanRewriter : public StmtExprMutator {
     return MergeNest(nest, body);
   }
   // Remap the index
-  PrimExpr RemapIndex(DataType dtype, PrimExpr index, StorageEntry* e) {
+  PrimExpr RemapIndex(PrimExpr index, StorageEntry* e) {
```

Comment:
Any reason we removed the dtype here? perhaps we can limit the lanes

Reply:
In my understanding the most correct dtype to use is the buffer data type itself. The `dtype` param before is load_op->dtype or store_op->dtype from context.
If the buffer dtype has non-trivial lanes like `tir.allocate([n], "float32*4")` (I'm not sure if it is possible), it seems not clear which factor to use based on original dtype param.

Reply:
In the current setup(our code generation), the Load and Store follows the dtype specified by the instruction, regardless of the dtype specified in allocation. So it is better to use dtype in the Node. But you are right that we should not multiply by lanes when calculating the offset

Reply:
Yes you're right! I checked some mixed dtypes and found that the type casting is done before indexing for load/store. I repush the change which modify just one line of code about lanes, can you kindly review it again?